flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -575,7 +585,7 @@ private RocksDB openDB(
 		try {
 			dbRef = RocksDB.open(
 				Preconditions.checkNotNull(dbOptions),
-				Preconditions.checkNotNull(path),
+				Preconditions.checkNotNull(Utils.getPathAsString(new Path(path))),
```

Comment:
Nit: Like this, the null check does not ensure that `path`is non-null, but the return value.

Reply:
If Path is null or "" then new Path(path) will throw runtime IllegalArgumentException. The path returned by getPathAsString will be checked again to be non-null.